.github/workflows/ci.yml

```
@@ -6,89 +6,280 @@ concurrency:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download test artifacts
+        uses: actions/download-artifact@37930b1c2abaa49bbe596cd826c3c89aef350131 # v7.0.0
+        with:
+          name: coverage-reports
+          path: ci-artifacts/
+        continue-on-error: true
+
+      - name: Commit test artifacts to PR branch
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        run: |
```

actionlint (1.7.10)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)